docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripDataTable.cs

```diff
@@ -0,0 +1,246 @@
+                        case "System.Char":
+                        case "System.String":
+                            jsonWriter.WriteString(key, valueString);
+                            break;
+                        case "System.Boolean":
+                            Boolean.TryParse(valueString, out bool boolValue);
+                            jsonWriter.WriteBoolean(key, boolValue);
+                            break;
+                        case "System.DateTime":
+                            if (DateTime.TryParse(valueString, out DateTime dateValue))
+                            {
+                                jsonWriter.WriteString(key, dateValue);
+                            }
+                            else
+                            {
+                                jsonWriter.WriteString(key, "");
+                            }
+                            break;
+                        case "System.TimeSpan":
+                            if (DateTime.TryParse(valueString, out DateTime timeSpanValue))
+                            {
+                                jsonWriter.WriteString(key, timeSpanValue.ToString());
+                            }
+                            else
+                            {
+                                jsonWriter.WriteString(key, "");
+                            }
+                            break;
+                        case "System.Byte":
+                        case "System.SByte":
+                        case "System.Decimal":
+                        case "System.Double":
+                        case "System.Single":
+                        case "System.Int16":
+                        case "System.Int32":
+                        case "System.Int64":
+                        case "System.UInt16":
+                        case "System.UInt32":
+                        case "System.UInt64":
+                            if (long.TryParse(valueString, out long intValue))
+                            {
+                                jsonWriter.WriteNumber(key, intValue);
+                            }
+                            else
+                            {
+                                double.TryParse(valueString, out double doubleValue);
+                                jsonWriter.WriteNumber(key, doubleValue);
+                            }
+                            break;
+                        default:
+                            jsonWriter.WriteString(key, valueString);
+                            break;
+                    }
+                }
+                jsonWriter.WriteEndObject();
+            }
+            jsonWriter.WriteEndArray();
+        }
+    }
+    public static class Extensions
+    {
+        public static DataTable JsonElementToDataTable(this JsonElement dataRoot)
+        {
+            var dataTable = new DataTable();
+            var firstPass = true;
+            foreach (var element in dataRoot.EnumerateArray())
```

Comment:
nit: use type name instead of `var` since it isn't stated on the RHS (here and elsewhere).